commands/template_pull_stack.go

```diff
@@ -98,14 +94,3 @@ func findTemplate(templateInfo []stack.TemplateSource, customName string) (speci
 	}
 	return nil
 }
-
-func pullSpecificTemplate(templateInfo []stack.TemplateSource, customName string, cmd *cobra.Command) error {
```

Comment:
Should this functionality really be removed? How should this behave when `regex` or `filter` is passed to the build?   These are persistent flags that are supposed to match function names in the yaml file

https://github.com/openfaas/faas-cli/blob/83081ef8ff0dbc2c8fb53954188a0f0f1f7e8e0f/commands/faas.go#L63-L64

Should the pull stack then ensure that it pulls the templates required for the matching functions?  What is the template folder exists but is missing a new template for a function that was appended to the file.  Perhaps pulling a single template is still a good idea

Reply:
Hey Lucas, interesting questions.

The first cluster of questions, the regex and filter are just skipped for the template pull stack. We handle configuration field only for the new functionality. Those flags are not handled in `pull stack` so they will have no affect on pulling.

For the second wave it will just pull the missing template. So if I have 3 templates and add 4th, try pulling, the 3 templates will be cached and won't be pulled, but the 4th one will be pulled as it does not exist.

I synced with Alex about this, he requested it in order for the feature to be more simple/elegant. @alexellis WDYT? Should we leave the feature, or remove it?

Reply:
So I ran this test
```
...
  newfuncyaw:
    lang: perl-alpine
    handler: ./newfuncyaw
    image: martindekov/newfuncyaw:latest
  saws:
    lang: python
    handler: ./saws
    image: martindekov/saws:latest

configuration:
  templates:
    - name: rusta
      source: https://github.com/openfaas-incubator/openfaas-rust-template
    - name: vala-http
    - name: crystal-http
```
`faas-cli build --regex="saws"`, the function build sucessfully, even though the first function did not have the `perl-alpine` template pulled(it would have got error.):
```
sudo ../../openfaas/faas-cli/faas-cli build --regex="saws"
Pulling template: rusta from configuration file: stack.yml
Fetch templates from repository: https://github.com/openfaas-incubator/openfaas-rust-template at master
2019/11/16 13:32:12 Attempting to expand templates from https://github.com/openfaas-incubator/openfaas-rust-template
2019/11/16 13:32:13 Cannot overwrite the following 1 template(s): [rust]
2019/11/16 13:32:13 Fetched 0 template(s) : [] from https://github.com/openfaas-incubator/openfaas-rust-template
Pulling template: vala-http from configuration file: stack.yml
Fetch templates from repository: https://github.com/affix/openfaas-templates-affix/ at master
2019/11/16 13:32:13 Attempting to expand templates from https://github.com/affix/openfaas-templates-affix/
2019/11/16 13:32:15 Cannot overwrite the following 4 template(s): [lua53 swift vala-valum vala-valum-http]
2019/11/16 13:32:15 Fetched 0 template(s) : [] from https://github.com/affix/openfaas-templates-affix/
Pulling template: crystal-http from configuration file: stack.yml
Fetch templates from repository: https://github.com/koffeinfrei/crystal-http-template at master
2019/11/16 13:32:15 Attempting to expand templates from https://github.com/koffeinfrei/crystal-http-template
2019/11/16 13:32:16 Cannot overwrite the following 1 template(s): [crystal-http]
2019/11/16 13:32:16 Fetched 0 template(s) : [] from https://github.com/koffeinfrei/crystal-http-template
[0] > Building saws.
Clearing temporary build folder: ./build/saws/
Preparing ./saws/ ./build/saws//function
Building: martindekov/saws:latest with python template. Please wait..
Sending build context to Docker daemon  8.192kB
Step 1/29 : FROM openfaas/classic-watchdog:0.18.1 as watchdog
 ---> 94b5e0bef891
Step 2/29 : FROM python:2.7-alpine
 ---> 1bf48bb21060
Step 3/29 : ARG ADDITIONAL_PACKAGE
 ---> Using cache
 ---> b683fd89484d
Step 4/29 : COPY --from=watchdog /fwatchdog /usr/bin/fwatchdog
...
```
Hope I got your right this time? The flow of the program can be traced, the building starts after pulling the templates, and the `regex` command will just make it so the function is skipped(if template for the skipped function is missing it will just not start building it, so there would be no error and the regexed function will be built with no problems).